app/src/main/java/com/example/tasky/feature_agenda/domain/use_case/Logout.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.tasky.feature_agenda.domain.use_case
+
+import com.example.tasky.feature_agenda.domain.repository.AgendaRepository
+import com.example.tasky.feature_authentication.domain.util.UserPreferences
+import com.example.tasky.util.Resource
+import com.example.tasky.util.Result
+
+class Logout(
+    private val repository: AgendaRepository,
+    private val userPrefs: UserPreferences
+) {
+
+    suspend operator fun invoke(): Result<Unit> {
+        val result = repository.logout()
+
+        return when(result) {
+            is Resource.Error -> Result.Error(result.message ?: "Unknown Error")
+            is Resource.Success -> {
+                userPrefs.clearPreferences()
+                Result.Success()
+            }
+            else -> Result.Error(result.message ?: "Unknown Error")
+        }
+    }
```

Comment:
This is fine to keep IMO as it contains real logic. You'd need to also clear the cache here, so that if the user logs in with a different account they don't see the agenda items from previously.